File: src/main/java/com/wesker926/leetcode/algorithms/p1576/s1/solution.go
// Package s1
// @author: wesker.gh
// @date: 2022/1/5
// @description: 遍历 + 替换
package s1

func modifyString(s string) string {
	cs := []rune(s)
	for i := 0; i < len(cs); i++ {
		if cs[i] != '?' {
			continue
		}
		for c := 'a'; c < 'd'; c++ {
			if (i != 0 && cs[i-1] == c) || (i != len(cs)-1 && cs[i+1] == c) {
				continue
			}
			cs[i] = c
			break
		}
	}
	return string(cs)
}

File: src/main/java/com/wesker926/leetcode/algorithms/p1576/s1/Solution.java
package com.wesker926.leetcode.algorithms.p1576.s1;

/**
 * @author wesker.gh
 * @date 2022/1/5
 * @description 遍历 + 替换
 */
public class Solution {
    public String modifyString(String s) {
        char[] cs = s.toCharArray();
        for (int i = 0; i < cs.length; i++) {
            if (cs[i] != '?') {
                continue;
            }

            for (char r = 'a'; r < 'd'; r++) {
                if ((i != 0 && cs[i - 1] == r) || (i != cs.length - 1 && cs[i + 1] == r)) {
                    continue;
                }
                cs[i] = r;
                break;
            }
        }
        return new String(cs);
    }
}
